src/projections/isea.cpp

```diff
@@ -1101,9 +1121,426 @@ PJ *PJ_PROJECTION(isea) {
+ * however slightly distorted. The angular distortion is below 17.27 degree, and
+ * the scale variation is less than 16.3 per cent.
+ *
+ * The projection has been proposed and has been described in detail by:
+ *
+ * John P. Snyder: An equal-area map projection for polyhedral globes.
+ * Cartographica, 29(1), 10–21, 1992. doi:10.3138/27H7-8K88-4882-1752
+ *
+ * Another description and improvements can be found in:
+ *
+ * Erika Harrison, Ali Mahdavi-Amiri, and Faramarz Samavati: Optimization of
+ * inverse Snyder polyhedral projection. International Conference on Cyberworlds
+ * 2011. doi:10.1109/CW.2011.36
+ *
+ * Erika Harrison, Ali Mahdavi-Amiri, and Faramarz Samavati: Analysis of inverse
+ * Snyder optimizations. In: Marina L. Gavrilova, and C. J. Kenneth Tan (Eds):
+ * Transactions on Computational Science XVI. Heidelberg, Springer, 2012. pp.
+ * 134–148. doi:10.1007/978-3-642-32663-9_8
+ */
+
+#include <algorithm>
+
+#define Min std::min
+#define Max std::max
+
+#define inf std::numeric_limits<double>::infinity()
+
+// distortion
+// static double maximumAngularDistortion = 17.27;
+// static double maximumScaleVariation = 1.163;
+// static double minimumScaleVariation = .860;
+
+// Vertices of dodecahedron centered in icosahedron faces
+static const GeoPoint facesCenterDodecahedronVertices[numIcosahedronFaces] = {
+    {E_RAD, DEG_TO_RAD * -144},  {E_RAD, DEG_TO_RAD * -72},
+    {E_RAD, DEG_TO_RAD * 0},     {E_RAD, DEG_TO_RAD * 72},
+    {E_RAD, DEG_TO_RAD * 144},   {F_RAD, DEG_TO_RAD * -144},
+    {F_RAD, DEG_TO_RAD * -72},   {F_RAD, DEG_TO_RAD * 0},
+    {F_RAD, DEG_TO_RAD * 72},    {F_RAD, DEG_TO_RAD * 144},
+    {-F_RAD, DEG_TO_RAD * -108}, {-F_RAD, DEG_TO_RAD * -36},
+    {-F_RAD, DEG_TO_RAD * 36},   {-F_RAD, DEG_TO_RAD * 108},
+    {-F_RAD, DEG_TO_RAD * 180},  {-E_RAD, DEG_TO_RAD * -108},
+    {-E_RAD, DEG_TO_RAD * -36},  {-E_RAD, DEG_TO_RAD * 36},
+    {-E_RAD, DEG_TO_RAD * 108},  {-E_RAD, DEG_TO_RAD * 180}};
+
+// static define precision = DEG_TO_RAD * 1e-9;
+#define precision (DEG_TO_RAD * 1e-11)
+#define precisionPerDefinition (DEG_TO_RAD * 1e-5)
+
+#define AzMax (DEG_TO_RAD * 120)
+
+#define westVertexLon (DEG_TO_RAD * -144)
+
+namespace { // anonymous namespace
+
+struct ISEAFacePoint {
+    int face;
+    double x, y;
+};
+
+class ISEAPlanarProjection {
+  public:
+    explicit ISEAPlanarProjection(const GeoPoint &value)
+        : orientation(value), cosOrientationLat(cos(value.lat)),
+          sinOrientationLat(sin(value.lat)) {}
+
+    bool cartesianToGeo(const PJ_XY &inPosition,
+                        const isea_inverse_params &params, GeoPoint &result) {
+        bool r = false;
+        static const double epsilon =
+            2E-8; // NOTE: 1E-11 seems too small for forward projection
+                  // precision at boundaries
+        int face = 0;
+        PJ_XY position = inPosition;
+// sin(-60)
+#define sr -0.86602540378443864676372317075293618347140262690519
```

Comment:
This is more a general question, not a request for this PR.

Nowadays compilers are much better optimizing constants at compile time.
See https://godbolt.org/z/3KeGe6Psj (not using the last versions nor the aggressive optimizations in purpose)
What about not using a define, but a const variable?
```suggestion
const double sr = std::sin(-60 * DEG_TO_RAD);
```

Is there any drawback on that?

Reply:
> Is there any drawback on that?

As far as I can see in https://en.cppreference.com/w/cpp/numeric/math/sin, it is guaranteed to be constexpr only in C++26, so even if most popular compilers have anticipated that, we don't have a guarantee of compile time computation of the constant with our current C++11 baseline.